charts/esphome/0.0.1/questions.yaml

```diff
@@ -0,0 +1,182 @@
+          schema:
+            type: string
+            default: "IfNotPresent"
+            enum:
+              - value: "IfNotPresent"
+                description: "Only pull image if not present on host"
+              - value: "Always"
+                description: "Always pull image even if present on host"
+              - value: "Never"
+                description: "Never pull image even if it's not present on host"
+  # Update Policy
+  - variable: strategyType
+    group: "Container Image"
+    label: "Update Strategy"
+    schema:
+      type: string
+      default: "Recreate"
+      enum:
+        - value: "RollingUpdate"
+          description: "Create new pods and then kill old ones"
+        - value: "Recreate"
+          description: "Kill existing pods before creating new ones"
+
+  # Configure Time Zone
+  - variable: timezone
+    group: "Configuration"
+    label: "Timezone"
+    schema:
+      type: string
+      default: "Etc/UTC"
+      $ref:
+        - "definitions/timezone"
+  # Configure Enviroment Variables
+  - variable: environment
+    group: "Configuration"
+    label: "Environment Variables"
+    schema:
+      type: list
+      default: []
+      items:
+        - variable: environmentVariable
+          label: ""
+          schema:
+            type: dict
+            attrs:
+              - variable: name
+                label: "Name"
+                schema:
+                  type: string
+              - variable: value
+                label: "Value"
+                schema:
+                  type: string
+
+  # Enable Host Network
+  - variable: hostNetwork
+    group: "Network"
+    label: "Enable Host Network"
+    schema:
+      type: boolean
+      default: false
+  # Service Configuration
+  - variable: service
+    group: "Network"
+    label: "Configure Service"
+    schema:
+      type: dict
+      attrs:
+        - variable: type
+          label: "Service type"
+          schema:
+            type: string
+            default: "NodePort"
+            enum:
+              - value: "NodePort"
+                description: "NodePort"
+              - value: "ClusterIP"
+                description: "ClusterIP"
+            show_subquestions_if: "NodePort"
+            subquestions:
+              - variable: port
+                label: "Port configuration"
+                schema:
+                  type: dict
+                  attrs:
+                    - variable: port
+                      label: "container port"
+                      schema:
+                        type: int
+                        default: 6052
+                        editable: false
+                    - variable: nodePort
+                      label: "Node Port to expose for UI"
+                      schema:
+                        type: int
+                        min: 9000
+                        max: 65535
+                        default: 36052
+                        required: true
```

Comment:
Interesting, so with this we have both the nodeport and clusterIP options with common library?
If so we really need to port this to all common-based apps

Reply:
as far as I can tell, both of these works. But I've only done basic testing

Reply:
Would you mind looking into hiding just the nodePort option when ClusterIP is selected, instead of hiding both?

Reply:
Sure thing :)